### DISCUSSION TITLE
Help, how can I read temperature and humidity from a Raspberry Pi Pico development board?

### DISCUSSION THREAD

**Discussion by sqxieshuai** [microsoft/devicescript#450] [Q&A] [answered]
I'm a beginner, can someone help me please?

**Comment by pelikhan** [comment]
What sensor do you plan to use for temperature/humidity sensing?

**Comment by sqxieshuai** [comment]
@pelikhan Thanks for your prompt response. The Pico board actually has a built-in temperature sensor. Here's an example of how to read values from this sensor using Python.

```python
from machine import ADC
import time
adc = machine.ADC(4) 
while True:
    ADC_voltage = adc.read_u16() * (3.3 / (65535))
    temperature_celcius = 27 - (ADC_voltage - 0.706)/0.001721
    temp_fahrenheit=32+(1.8*temperature_celcius)
    print("Temperature: {}°C {}°F".format(temperature_celcius,temp_fahrenheit))
    time.sleep_ms(500)
```

how can I achieve similar functionality using DeviceScript?

**Reply by pelikhan** [reply]
Indeed! In section https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf 4.9, the temperature sensor is documented. I don't think it is accessible from DeviceScript GPIOs yet as it is an "internal" pin. This will require a change to devicescript-pico.

(the control service has a temperature register that needs to be wired up https://github.com/microsoft/jacdac-c/blob/c64f2c6f7571e34dcd8c30d755dfad640aa44ceb/source/jd_control.c#L142 )

**Reply by pelikhan** [reply]
I've created an issue to track this: https://github.com/microsoft/devicescript/pull/454

**Reply by pelikhan** [reply, marked as answer]
You can use 'mcuTemperature' in the latest build.
